examples-dot-net/README.md

```diff
@@ -0,0 +1,1128 @@
+            ArcExtensionResource arcExtension = client.GetArcExtensionResource(arcExtensionResourceId);
+```
+
+4. Invoke delete operation
+
+```C# Snippet: 
+            
+            arcExtension.DeleteAsync(WaitUntil.Completed);
+```
+
+5. For confirmation we can print the text "Succeeded"
+
+```C# Snippet: 
+        Console.WriteLine($"Succeeded");
+```
+
+#### Upgrade Machine Extensions
+
+1. Get azure token
+
+```C# Snippet: 
+            TokenCredential cred = new DefaultAzureCredential();
+            ArmClient client = new ArmClient(cred);
+```
+2. Update the parameters given below
+
+```C# Snippet:
+            string subscription = "00000000-0000-0000-0000-000000000000"; # Replace with your subscription ID
+            string resourceGroupName = "hcicluster-rg"; # Replace with your resource group name
+            string clusterName = "HCICluster"; # Replace with your cluster name
+            string arcSettingName = "HCIArcSettingName"; # Replace with your Arc Setting Name
+            string extensionName = "MicrosoftMonitoringAgent"; # Replace with your extension name
+```
+3. Invoke upgrade operation
+
+```C# Snippet: 
+            ExtensionUpgradeContent content = new ExtensionUpgradeContent()
+            {
+                TargetVersion = "1.0.18062.0",
+            };
+            await arcExtension.UpgradeAsync(WaitUntil.Completed, content);
+
+
+```
+
+4. For confirmation we print text "Succeeded"
+
+```C# Snippet: 
+        Console.WriteLine($"Succeeded");
+```
+
+### Arc Setting Resource
+
+#### Get a Arc Setting Resource
+
+1. Get the Azure token
+
+```C# Snippet: 
+            TokenCredential cred = new DefaultAzureCredential();
+            ArmClient client = new ArmClient(cred);
+```
+2. Update the parameters given below
+
+```C# Snippet:
+            string subscription = "00000000-0000-0000-0000-000000000000"; # Replace with your subscription ID
+            string resourceGroupName = "hcicluster-rg"; # Replace with your resource group name
+            string clusterName = "HCICluster"; # Replace with your cluster name
+            string arcSettingName = "HCIArcSettingName"; # Replace with your Arc Setting Name
+```
+3. Get Arc Extension Resource
+
+```C# Snippet: 
+            ResourceIdentifier arcSettingResourceId = ArcSettingResource.CreateResourceIdentifier(subscriptionId, resourceGroupName, clusterName, arcSettingName);
+            ArcSettingResource arcSetting = client.GetArcSettingResource(arcSettingResourceId);
+            // the variable result is a resource, you could call other operations on this instance as well
+             ArcExtensionResource result = arcExtension.GetAsync().Result;
+
+
+```
+
+4. For confirmation we will print the id retrieved from result
+```C# Snippet: 
+        ArcExtensionData resourceData = result.Data;
+        Console.WriteLine($"Succeeded on id: {resourceData.Id}");
+```
+
+#### Update Arc Setting Resource
```

Comment:
Instead of adding this separately. 
Add it as part of installing WAC. For WAC, we need to enable the connectivity first